qiskit_nature/second_q/formats/fcidump/fcidump.py

```diff
@@ -0,0 +1,130 @@
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""FCIDump"""
+
+from __future__ import annotations
+
+from typing import List
+from dataclasses import dataclass
+from pathlib import Path
+import numpy as np
+
+from qiskit_nature import QiskitNatureError
+
+
+from .dumper import _dump_1e_ints, _dump_2e_ints, _write_to_outfile
+from .parser import parse
+
+
+@dataclass
+class FCIDump:
+    """
+    Qiskit Nature dataclass for reading an FCIDump file.
+
+    The FCIDump format is partially defined in Knowles1989.
+
+    References:
+        Knowles1989: Peter J. Knowles, Nicholas C. Handy,
+            A determinant based full configuration interaction program,
+            Computer Physics Communications, Volume 54, Issue 1, 1989, Pages 75-83,
+            ISSN 0010-4655, https://doi.org/10.1016/0010-4655(89)90033-7.
+    """
+
+    hij: np.ndarray | None
+    """The pair of alpha 1-electron integrals."""
+    hij_b: np.ndarray | None
+    """The pair of beta 1-electron integrals."""
+    hijkl: np.ndarray | None
+    """The triplet of alpha/alpha 2-electron integrals."""
+    hijkl_ba: np.ndarray | None
+    """The triplet of beta/alpha 2-electron integrals."""
+    hijkl_bb: np.ndarray | None
+    """The triplet of beta/beta 2-electron integrals."""
+    multiplicity: int
+    """The multiplicity."""
+    num_electrons: int
+    """The number of electrons."""
+    num_orbitals: int
+    """The number of orbitals."""
+    nuclear_repulsion_energy: float | None
+    """The nuclear repulsion energy"""
+    orbsym: List[str] | None
+    """A list of spatial symmetries of the orbitals."""
+    isym: int
+    """The spatial symmetry of the wave function."""
+
+    @classmethod
+    def from_file(cls, fcidump: str | Path) -> FCIDump:
+        """Constructs an FCIDump object from a file."""
+        data = parse(fcidump if isinstance(fcidump, Path) else Path(fcidump))
+        return cls(
+            hij=data.get("hij", None),
+            hij_b=data.get("hij_b", None),
+            hijkl=data.get("hijkl", None),
+            hijkl_ba=data.get("hijkl_ba", None),
+            hijkl_bb=data.get("hijkl_bb", None),
+            multiplicity=data.get("MS2", 0) + 1,
+            num_electrons=data.get("NELEC"),
+            num_orbitals=data.get("NORB"),
+            nuclear_repulsion_energy=data.get("ecore", None),
+            orbsym=data.get("ORBSYM", None),
+            isym=data.get("ISYM"),
+        )
+
+    def to_file(self, fcidump: str | Path) -> None:
+        """Dumps an FCIDump object to a file.
+
+        Args:
+            fcidump: Path to the output file.
+        Raises:
+            QiskitNatureError: invalid number of orbitals.
+        """
+        outpath = fcidump if isinstance(fcidump, Path) else Path(fcidump)
+        # either all beta variables are None or all of them are not
```

Comment:
That method was small enough that I just moved ot the other one. I didn't change the other methods, so I kept there not to put a lot of code in the fcidump file and leave it with the kernel of a dataclass.

Reply:
I was thinking that we could maybe make the `parse` method in the `parser` utility file, also private. But I guess it does not really matter much, since its not included in the docs anyways